resources/test/tests/unit/basic.html

```diff
@@ -24,7 +24,12 @@
         assert_equals(typeof WebIDL2.parse, "function");
     }, 'WebIDL2 namespace should have a parse method');
     test(function() {
-        assert_throws(new TypeError(), function() {  WebIDL2.parse("I'm a syntax error"); });
```

Comment:
Why rewrite this to not use `assert_throws`?

Reply:
(Like, yes, the error type has changed, but can we not still use `assert_throws`?)

Reply:
`assert_throws` seems to allow only DOMException and JS native error objects that have `name` property. Should the function be fixed to allow arbitrary error objects?

Reply:
Made https://github.com/web-platform-tests/wpt/issues/11556 for this